File: app/src/main/java/io/particle/android/sdk/tinker/TinkerPrefs.java
package io.particle.android.sdk.tinker;

import android.content.Context;
import android.content.SharedPreferences;

class TinkerPrefs {

    private static final String BUCKET_NAME = "tinkerPrefsBucket";
    private static final String KEY_IS_VISITED = "isVisited";

    private static TinkerPrefs instance = null;


    static TinkerPrefs getInstance(Context ctx) {
        if (instance == null) {
            instance = new TinkerPrefs(ctx);
        }
        return instance;
    }


    private final SharedPreferences prefs;


    private TinkerPrefs(Context context) {
        prefs = context.getApplicationContext()
                .getSharedPreferences(BUCKET_NAME, Context.MODE_PRIVATE);
    }

    boolean isFirstVisit() {
        return !prefs.getBoolean(KEY_IS_VISITED, false);
    }

    void setVisited(boolean isVisited) {
        prefs.edit().putBoolean(KEY_IS_VISITED, isVisited).apply();
    }

}

File: app/src/main/java/io/particle/android/sdk/tinker/TinkerPrefs.kt
package io.particle.android.sdk.tinker

import android.annotation.SuppressLint
import android.content.Context
import android.content.SharedPreferences


private const val BUCKET_NAME = "tinkerPrefsBucket"
private const val KEY_IS_VISITED = "isVisited"
private const val KEY_API_BASE_URI = "KEY_API_BASE_URI"


internal class TinkerPrefs private constructor(context: Context) {

    companion object {

        @SuppressLint("StaticFieldLeak")  // we use app context, so it doesn't matter
        private var instance: TinkerPrefs? = null

        fun getInstance(ctx: Context): TinkerPrefs {
            var inst = instance
            if (inst == null) {
                inst = TinkerPrefs(ctx)
                instance = inst
            }
            return inst
        }
    }

    private val app: Context = context.applicationContext
    private val prefs: SharedPreferences =
        app.getSharedPreferences(BUCKET_NAME, Context.MODE_PRIVATE)

    val isFirstVisit: Boolean
        get() = !prefs.getBoolean(KEY_IS_VISITED, false)

//    val apiBaseUri: String?
//        get() = _getApiBaseUri()

    fun setVisited(isVisited: Boolean) {
        prefs.edit().putBoolean(KEY_IS_VISITED, isVisited).apply()
    }

//    private fun _getApiBaseUri(): String? {
//        return
//    }

}
